app/src/main/java/chat/rocket/android/chatrooms/presentation/ChatRoomsPresenter.kt

```diff
@@ -141,6 +144,36 @@ class ChatRoomsPresenter @Inject constructor(
         }
     }
 
+    fun toggleFavoriteChatRoom(roomId: String, isFavorite: Boolean) {
+        launchUI(strategy) {
+            try {
+                // Note that if it is favorite then the user wants to unfavorite - and vice versa.
+                retryIO("favorite($roomId, $isFavorite)") {
+                    client.favorite(roomId, !isFavorite)
+                }
+//                view.showFavoriteIcon(!isFavorite)
```

Comment:
We need to show/hide that icon after user interaction.

Reply:
The bottom sheet will disappear after toggle, and next time it will fetch from toggled data. So i guess, no need of `showFavoriteIcon`

Reply:
But are this PR adding these features from the chat list or inside a chat room?

Reply:
From chatList, actually, I was taking refernce of code implemented in ChatRoom, thats why these functions are there. I will remove that.

Reply:
No! Don't remove that, please! Give the user the possibility to star/unstar a chat inside a chat room (like we do for the web for instance).

You should leave the current feature as it because removing that will force the user to go back to the chat list to do this actions everytime.

Reply:
No, I am not removing the current feature.
These functions are implemented in Chatlist( not chatRoom). So it doesn't change the feature of Chatroom.

Reply:
Ah, ok 😃 Then we shouldn't put this function neither on ChatRoomsPresenter.kt nor ChatRoomPresenter.